src/solver/surfaceIntegrations.F90

```diff
@@ -1359,6 +1436,111 @@ subroutine computeCpMinFamily(famList)
 
     end subroutine computeCpMinFamily
 
+    subroutine computeSepSenMaxFamily(famList)
+
+        use constants
+        use inputTimeSpectral, only: nTimeIntervalsSpectral
+        use communication, only: ADflow_comm_world, myID
+        use blockPointers, only: nDom
+        use inputPhysics, only: sepSenMaxFamily, velDirFreeStream
+        use blockPointers
+        use flowVarRefState
+        use inputCostFunctions, only: sepsensorksphi
+        use BCPointers
+        use utils, only: setPointers, setBCPointers, isWallType, EChk
+        use sorting, only: famInList
+
+        implicit none
+
+        integer(kind=intType), dimension(:), intent(in) :: famList
+        integer(kind=intType) :: mm, nn, sps
+        integer(kind=intType) :: i, j, ii, blk, ierr
+        real(kind=realType) :: sepsensor_local
+        real(kind=realType) :: vectTangential(3), v(3)
+        real(kind=realType) :: vectDotProductFsNormal, sensor
+
+        ! this routine loops over the surface cells in the given family
+        ! and computes the true maximum sepsensor value.
+        ! this is then used in the surface integration routine to compute
+        ! the maximum sepsensor value using KS aggregation.
+        ! the goal is to get a differentiable maximum sepsensor output.
+
+        ! loop over the TS instances and compute sepSenMaxFamily for each TS instance
+        do sps = 1, nTimeIntervalsSpectral
+            ! set the local sepsensor to a smaller value so that we get the actual max
+            sepsensor_local = -10000.0_realType
+            do nn = 1, nDom
+                call setPointers(nn, 1, sps)
+
+                ! Loop over all possible boundary conditions
+                bocos: do mm = 1, nBocos
+                    ! Determine if this boundary condition is to be incldued in the
+                    ! currently active group
+                    famInclude: if (famInList(BCData(mm)%famID, famList)) then
```

Comment:
This part of the code has a lot of nesting, which I consider bat practice because it makes it hard to read. I prefer if you invert the if-checks and `cycle` the loop. E.g.:
```
if (.not. famInList(BCData(mm)%famID, famList)) then
    cycle
end if
call setBCPointers(mm, .True.)
...
```

But you could also argue that ADflow is full of nesting and this instance does not really matter in the grand scheme of things...